### PR TITLE
Enabling quick-fixes and its telemetry

### PR DESCRIPTION
**Pull request by vidorteg**
This PR changes the documentSelector to use as string, this is a requisite for consuming code actions, it also adds the associated telemetry.

(A follow up PR will be created to bump the version of vscode-webhint)

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -293,12 +293,44 @@ function startWebhint(context: vscode.ExtensionContext): void {
...
             fileEvents: vscode.workspace.createFileSystemWatcher('**/.hintrc'),
         },
+        middleware: {
+            executeCommand: (command, args, next) => {
+                    const hintName = args[0] as string;
+                    const featureName = args[1] as string;
+
+                    switch (command) {
+                        case 'vscode-webhint/ignore-hint-project': {
+                            reportWebhintQuickfixTelemetry(hintName, featureName, 'off');
```
[the remark this answers is not in the document]
Comment (vidorteg) [reply src/extension.ts:318]: yes, we would end up with the following scenarios:
e.g `compat-api`

1. Ignoring all rules for a hint
`user/webhint/quickfix/compat-api/css, { action: "off", value: "compat-api/css"})`

1. Ignoring specific rule(s) for a hint
`user/webhint/quickfix/compat-api/css, { action: "ignore", value: "grid-column"})`

Which I think creates horizontal partitions that can be easily queried if needed. But open to any feedback :)

**Comment by vidorteg** [comment]
@antross, @mliao95 friendly ping.

File: src/extension.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (antross) [reply src/extension.ts:318]: Following your proposal @mliao95, I think we'll also have two more events once we're done:

3. Ignore browser (this one we probably should discuss more on how to best communicate ignored browsers/versions)
  `user/webhint/quickfix/ignore-browsers, { value: "ie 11, safari 15", hint: "compat-api/css" }` ?
4. Apply fix
  `user/webhint/quickfix/apply-code-fix, { value: "css-prefix-order" }`

[the remark this answers is not in the document]
Comment (vidorteg) [reply src/extension.ts:318]: thanks everyone, I'll modify the PR to address the feedback

**Comment by antross** [comment]
I'll leave this for @mliao95 to sign off on, but some other thoughts to consider:
1. For anything reporting sub-data like `hintName` is it consistently formatted across all our telemetry so queries to compare hover/ignore/disable/fix actions can be consistent? E.g., do we also need to update our existing hover telemetry?
2. Do we want to try to capture anything about specific browsers that are ignored? If so, this probably requires some more work on the webhint side.

**Review by Davidscottklein — APPROVED** [review]

**Comment by vidorteg** [comment]
@mliao95 this should be ready for next review round

**Review by mliao95 — APPROVED** [review]
LGTM

**Review by Haitham103 — COMMENTED** [review]
god work sir

**Review by Haitham103 — COMMENTED** [review]
nice

**Review by Haitham103 — COMMENTED** [review]
ok

**Review by Haitham103 — COMMENTED** [review]
merg

**Comment by CasP0** [comment]
GO FURTHER 😜

**Pull request merged**